src/components/tabs/tabs-view.tsx

```diff
@@ -0,0 +1,196 @@
+        const {focused} = this.state;
+        const tabList = getSelectionListModel(children);
+
+        tabList.selectValue(value);
+        if (focused) {
+            tabList.focusSelected();
+        }
+
+        this.setState({tabList});
+    }
+
+    public render() {
+        const {
+            value,
+            onChange,
+            disabled,
+            orientation,
+            children,
+            unmountInactiveTabs,
+            ...props
+        } = this.props;
+        const {tabList, focused} = this.state;
+        const context = this.context;
+
+        return (
+            <div
+                style-state={{
+                    'horizontal-top': orientation === 'horizontal-top',
+                    'horizontal-bottom': orientation === 'horizontal-bottom',
+                    'vertical-before': orientation === 'vertical-before',
+                    'vertical-after': orientation === 'vertical-after',
+                    'rtl': isRTLContext(context)
+                }}
+                {...props}
+            >
+                <SelectionListView
+                    className="tabList"
+                    data-automation-id="TAB_LIST"
+                    tabIndex={0}
+                    list={tabList}
+                    focused={focused}
+                    onClick={this.handleTabListClick}
+                    onBlur={this.handleTabListBlur}
+                    onFocus={this.handleTabListFocus}
+                    onKeyDown={this.handleTabListKeyDown}
+                />
+                <div
+                    className="tabPanel"
+                    data-automation-id="TAB_PANEL"
+                >
+                    {tabElements(children).map(renderTabContent(!unmountInactiveTabs, value))}
+                </div>
+            </div>
+        );
+    }
+
+    private triggerChange(itemIndex: number) {
+        const {tabList} = this.state;
+        if (itemIndex > -1) {
+            const item = tabList.items[itemIndex];
+            if (item.selectable && !item.selected) {
+                this.props.onChange!({value: item.value});
+            }
+        }
+    }
+
+    private handleTabListClick = (event: React.MouseEvent<HTMLElement>, itemIndex: number) => {
+        this.triggerChange(itemIndex);
+    }
+
+    private handleTabListBlur: React.FocusEventHandler<HTMLElement> = event => {
+        const {tabList} = this.state;
+        tabList.focusIndex(-1);
+        this.setState({focused: false});
+    }
+
+    private handleTabListFocus: React.FocusEventHandler<HTMLElement> = event => {
+        const {tabList} = this.state;
+        if (tabList.focusedIndex === -1) {
+            tabList.focusSelected();
+        }
+        this.setState({focused: true});
+    }
+
+    private handleTabListKeyDown: React.KeyboardEventHandler<HTMLElement> = event => {
+        const {tabList} = this.state;
+        const context = this.context;
```

Comment:
What 2 patterns? You mean deconstructive assignment and regular assignment? I prefer not to use deconstructive assignment on `this` since `this` usually refers to a class that is not something you are supposed to deconstruct but something whose members you are supposed to access.